src/server/server.py

```diff
@@ -4425,6 +4449,11 @@ def initialize_rh_interface():
                 except ImportError:
                     logger.warning("Unable to import libraries for I2C nodes; try:  " +\
                                    "sudo pip install --upgrade --no-cache-dir -r requirements.txt")
+                    ui_server_messages['i2c'] = {
+                        'state': 'library',
+                        'header': __('Warning'),
+                        'message': __("Unable to import libraries for I2C nodes. Try: <code>sudo pip install --upgrade --no-cache-dir -r requirements.txt</sudo>")
+                    }
```

Comment:
I think you should have a function that does this

Reply:
Do you mean something like add_ui_message that takes what is set here as arguments? What's the major benefit to that approach?

Reply:
You're specifying the same dictionary names ('state'/'header'/'message') in several places. If one happens to be misspelled it can be a difficult error to track down. You're loading the data in the same, specific way in each place, so better to put the loading into a single function. Also, better to avoid accessing global variables as much as possible; you can put most of the accesses to 'ui_server_messages' into the single function.

Reply:
Thanks, I appreciate it! See recent commit.

Reply:
Looks great.